### ISSUE TITLE
[Feature] Make sankey diagram dragable and be able to zoom in/out

### ISSUE THREAD

**Issue by SincerelyUnique** [apache/echarts#19863]
### What problem does this feature solve?

This feature can make sankey diagram dragable and be able to zoom in and out, for the sankey with a large dataset, the picture will be overflow to current div, so the current screen unable to handle all data , we want to make the canvas canbe dragable and can be zoom in or zoom out

### What does the proposed API look like?

maybe like a attribute like dataZoom: {} or some others
Thanks team!

**Comment by SincerelyUnique** [comment]
maybe can be implement from 

https://blog.csdn.net/ztwelve/article/details/128649174

but it is not smooth

I prefer this one: https://codepen.io/borntofrappe/pen/KrBypK

but I need to learn d3, it is sad

**Comment by helgasoft** [comment]
Interesting examples. Could be done eventually in ECharts, but needs work - [Demo](https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWtJgDYFMBcSYOAHmHgOQC2wArgM44DKAxgE7BZawBmwrsAL2DBK5AL4AaNLAasIOOnlgBtaelTpNGAJ4h8scnQCG0ANY5t5KVvRYj22mQPQYOK2s05KIABZG6EIrIHlq8zPRK5EYAJgBWRsw40MyWIbCSadFGYEZKqjaaGgWa0EaU-lHkaegZxcF16KXlkQBGVXW1xUV1TRVGAIztxZ0F3cW9kUYATEMFIzZjBRMGLYPV6dZd641lFcyzNmJpALqbWlgQZkH5Ww0ytKyJk-53OawA5jhOUYNnxQBuRiwNH0AFZ1vMtIsbHQHk8DEYXg03p9vtMkXVAcD9ABmP4FOwtHBYJRIGQ-YAAdyUYFYIPSEPxhW29xojwqbSZWhRX1avxZWJBSgAHFzNITiaTyVSaXScAyOmLoVpYWz4ZUxeged9VhiAUChbA8SyJSTkNLqRg5QrhkqWar2Xy9QVtZN-XdBfp-prYKapXQKZbafSjor1srNA71bqfa6DPsfZ6lFMfX7zQGZVaQxCTh55mTYGkAPRF2CXf6BCAtXCwd6sIy-DDAWAgdgV6LyusNnwAJWAZVgOH-STAaTAugqXd8zt99kckRc0DcTLo2koLQ4jAgAi9AAZ90yIERKIxx7gpaAEkftEpdxs0uwyrKQUysjk8uGWctyAA5YAd2B3QaYhbx9G9YF3RlPzub8_wAmYfRAwD913MClH6FCcwKY482kHCxAAbhQFBmBgOgwEEfhEEobQAGE_FYMAADpUQALVYAAKABKIiaPooxGKYmAOPIKde37UQJFgDicC4hAAD56lgUjoHIwRhEoBBByYoQRCI9AIG4aTdMoOSxkBfhiToLSBFYJjyL4IxPhYr4ABFAhAOxtAAGUCMBuP0zwsDoJjeFYABRBIfA46TiTk-BFOhQzYqweyzxwISQCvcdYEU3czPWYkmOyWkRLoZggWXFQTKkkzjh4tJQ00MQGpqFAWoIoA)

**Comment by jborch** [comment]
I solved it this way, works quite nicely:

https://echarts.apache.org/examples/en/editor.html?c=sankey-nodeAlign-left&lang=ts&code=JYOwLgpgTgZghgYwgAgOJWAE2QbwFDLJgD2ADgFzIgCuAtgEbQDcByGA5gBZiU0POt6xMCVq86jKC0IAbCDB5UJAgL548oSLEQoAWsWK0AQsQAeuVqYDKYOFEV9J05KYCiITOP5TWATxt2Dso-hL7unkreLGp4MNQgCGDAxCDI7BBg-oboWAAUrADOwABeEJQ4yADuWGCcXk7InBDAXEHeyCoANKzFBsZmlFn9pt2EmPJw1DJgORGzeACUlLMWhAgpBWC4LgH2nS7h-_62e8hhHh3IALzIvYYmpiyswDDIudYnW1ffyAAMyAAyAEHC7fG4ARl-_yBZ12Xx-0OB52wYOQkN-C1WhDYGWoUFS43gUxmGEwzhiaw2WzgADc4MAZHB6HIAOo1TjXZBFUoAOmqmFqyAAtMhCZNprMeXIFMLRRNiZKONxnOsQJtkLT6YzmRAABLNVqc7kQHlNFrcWVihWknkkUiW-USm1CESGJ6UtVbAoIOByABiiBIUAAGpz0cgAPRvNwXEUfQILFVUrk-_2B4hQACaYahkbeyNlxwT7uQqvV3t9EFDNyFmoZTLk-vNWwAVG8K2nEhnsyLwYnWGWvamINma3Xtaz2cg27kOxAA12Q7K-yXB0QoHA1Yykilq9G4Xn0Zi2-OGxA2QKOW25wug8Gk5715uCtvkiBR29cuGRcjMVGj9OGp0vWOpNoa17Dre3ZMJGUYACpkMg0pbOs-LQMgwAFMgABEyLYfmGqmJhXKfFhMBQIYRBNMgLqiAs6geuqxAwDABQZAh9o1mAG5bnAO7vg-TEsWxYAADLyF8wrcc-r67oJWzMaxGQmK6tBGsOoYANRPrx_GZvJyCKSJABKzbqZW2badJulvveDE4mAeKpPg2JIRJlBWk6WBShJyDaUZGTiQoozYkqiieSS3lhX5hnCRkpmtCFhB2h5jqRZgtqIf5cVgBxSU0cIoipUSXkZbRlHZUpYAqaIrAqNE6i0L4ADCnCBDyBScMQlSicQcCYKA7C5P2eAACQ8ukYC5MZADyM1wQA-gACgAgnBuoxQA5BGmB8XAEZwAUIk7XtEYQCA0DsL4PIAFYFCkm37HECT8W8u22JiLnIE1rXtZwWAQL1_WDcNq7JhFszLKSnJfclZCUAAzL8-XIZQABMyOsIQYWUOCAAsmOueVYjIEjdUlj9bX2DyKS5Jt71wEMj1vJiVwAHxYoxWx3LQM2kK9NyU-1k18_xw0ANr03tTMALoalhm6-OLMvOFzLgrURWE3CtUAbtd5GGLkPOi2-Cw8jAoCYLk7yUIrrMc6YPKmBrmEAJIeBA5iohiqulsmvgu1ryA63r5sUbQRt9CbKRmxbHjW74tsgL49tnDyAea-74xewiI2uS8bwAITO5ryAAD5l8ghcZ5hn1Y6FuL4r7FLYmu7DQzck1DLM-SuYQQvU5NAAihtm5NJjxANIDsM1MjAOdYDGRAiTDflhCw338b2JQJeYR1nxr9iMYRLvBQ8udmCH6EcKUDXZ-bIEV9nOEt-B-fHj14QXSf3KJXpfX_ZXIDzAB1DI0cQC5A3oQNiGAIAFEoOLH-68kHYiwJQTaBRNwAGsIC-EeigwgPIiHtywCgtQfdkAy3rioQBHQ87fRalTEBIlwG92xLkMgAtObYiSGAOQ5QUGQFMIoTaAA5Yg4xg5z3YKkBK3BNo_2_hQkgxBpjAAoNw1y3EWjpCgOg4AkBaD4IoclDguiZogHQbQYg1A2LWJpBABRFClGb0DgIkxhFMKiVAGUTRfdOrdUoPAGQbEyGKMPnfdxJi4Ca28RdKJHiAmVCCb6UJHjyF9xcVogwMghCmASX3FKyBNq_GMSYmAEA-J4l8VAvuUA4FBhqQQ7ESTKDcWoBAJ-WiDH8OKUvTYGZHFdI6GQw-WTsQkIiBDUkh8GZDAQSgiMUYhiGVSC6DkMS4EoNqVo3wpBfGbVAEUcYZSPEW2mNAAAshIg5IAUhDOaafLOntKCEw8d9UAVhSCblxqM7ZzSwB7IOUcgGpzykMi0Nc8Y6C7kXTBRQu-zz8l_GGbQT53zLFojCRQmWh8YHz3gcgRBHidmoIiBg7BuD4WFKBegzBIAcF4OGRAWgpA2pFEJaS1yMBiAIFseg_qN0dAJDwc08ZFCiE8mmVgYZdzxgrWkZizayFqWuQZh5PaPI5VwOGXPBlhKGZSlAFggouqfE2F8L0rlrdVEZjpTYqASBVV9z5VABxF0jqvJ5AAVjFdivuVDXI0NYP2GhTAgA

**Comment by Hubbitus** [comment]
Indeed, the last implementation looks good, and it will be very cool to see it in the core library.

**Comment by dmitrijs-pavlovs-dev** [comment]
> I solved it this way, works quite nicely:
Link stopped working, do you still have it somewhere ?

**Comment by jborch** [comment]
> > I solved it this way, works quite nicely:
> > Link stopped working, do you still have it somewhere ?

The link still works for me - but here is the _code:
https://gist.github.com/jborch/d952728edfc285f4d3927eb55450db74
